src/test/java/io/camunda/testing/assertions/MessageAssertTest.java

```diff
@@ -0,0 +1,143 @@
+package io.camunda.testing.assertions;
+
+import static io.camunda.testing.assertions.BpmnAssert.assertThat;
+import static io.camunda.testing.util.Utilities.deployProcess;
+import static io.camunda.testing.util.Utilities.sendMessage;
+import static io.camunda.testing.util.Utilities.startProcessInstance;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import io.camunda.testing.extensions.ZeebeAssertions;
+import io.camunda.testing.util.Utilities.ProcessPackMessageEvent;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.PublishMessageResponse;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import org.camunda.community.eze.RecordStreamSource;
+import org.camunda.community.eze.ZeebeEngine;
+import org.camunda.community.eze.ZeebeEngineClock;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+@ZeebeAssertions
+class MessageAssertTest {
+
+  public static final String CORRELATION_KEY = "correlationkey";
+  public static final String WRONG_CORRELATION_KEY = "wrongcorrelationkey";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+  private ZeebeEngineClock clock;
+
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasBeenCorrelated() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackMessageEvent.RESOURCE_NAME);
+      final Map<String, Object> variables =
+          Collections.singletonMap(
+              ProcessPackMessageEvent.CORRELATION_KEY_VARIABLE, CORRELATION_KEY);
+      startProcessInstance(client, ProcessPackMessageEvent.PROCESS_ID, variables);
+
+      // when
+      final PublishMessageResponse response =
+          sendMessage(client, ProcessPackMessageEvent.MESSAGE_NAME, CORRELATION_KEY);
+
+      // then
+      assertThat(response).hasBeenCorrelated();
+    }
+
+    @Test
+    void testHasExpired() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackMessageEvent.RESOURCE_NAME);
+      final Duration timeToLive = Duration.ofNanos(1);
+
+      // when
+      final PublishMessageResponse response =
+          sendMessage(client, ProcessPackMessageEvent.MESSAGE_NAME, CORRELATION_KEY, timeToLive);
+      clock.increaseTime(timeToLive.plusMillis(1));
+
+      // then
+      assertThat(response).hasExpired();
+    }
+
+    @Test
+    void testExtractingProcessInstance() throws InterruptedException {
```

Comment:
:thought_balloon: I think the example is not very helpful to understand the purpose of this method. Would be better to have a process that is started by a message event. Then it would be clearer why this is useful

Reply:
I have added more tests for testing if a process is started. For this I had to look at a different set of records. I've also added 2 more assertions to verify whether a start event message has been correlated or not.